java.nio.file.AccessDeniedException during IO.createDirectory on Windows

## steps

Not sure but I'm getting this on scripted `classloader-cache/scalatest`? test on AppVeyor.

The stack trace goes back to `IO.createDirectory`.

## problems

https://ci.appveyor.com/project/sbt/sbt/builds/32726198

<details>
<summary>

```scala
[info] Running classloader-cache/scalatest
[error] java.io.IOException: Could not create directory C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src: java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src
[error] 	at sbt.io.IO$.createDirectory(IO.scala:337)
....
[error] Caused by: java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src
[error] 	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:83)
[error] 	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
[error] 	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
[error] 	at sun.nio.fs.WindowsFileSystemProvider.createDirectory(WindowsFileSystemProvider.java:504)
[error] 	at java.nio.file.Files.createDirectory(Files.java:674)
[error] 	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
[error] 	at java.nio.file.Files.createDirectories(Files.java:767)
[error] 	at sbt.io.IO$.$anonfun$createDirectory$1(IO.scala:333)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:40)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] 	at sbt.io.IO$.createDirectory(IO.scala:335)
```

</summary>


```scala
[info] Running classloader-cache/scalatest
[error] java.io.IOException: Could not create directory C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src: java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src
[error] 	at sbt.io.IO$.createDirectory(IO.scala:337)
[error] 	at sbt.io.IO$.copyImpl(IO.scala:835)
[error] 	at sbt.io.IO$.$anonfun$copy$1(IO.scala:826)
[error] 	at scala.Function$.$anonfun$tupled$1(Function.scala:79)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:273)
[error] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:273)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:266)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at sbt.io.IO$.copy(IO.scala:826)
[error] 	at sbt.io.IO$.copyDirectory(IO.scala:872)
[error] 	at sbt.io.IO$.copyDirectory(IO.scala:861)
[error] 	at sbt.io.IO$.copyDirectory(IO.scala:844)
[error] 	at sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$1(ScriptedTests.scala:245)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:273)
[error] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:273)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:266)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at sbt.scriptedtest.ScriptedTests.runBatchTests$1(ScriptedTests.scala:240)
[error] 	at sbt.scriptedtest.ScriptedTests.runBatchedTests(ScriptedTests.scala:291)
[error] 	at sbt.scriptedtest.ScriptedTests.$anonfun$batchScriptedRunner$10(ScriptedTests.scala:148)
[error] 	at sbt.io.IO$.withTemporaryDirectory(IO.scala:490)
[error] 	at sbt.io.IO$.withTemporaryDirectory(IO.scala:500)
[error] 	at sbt.scriptedtest.ScriptedTests.$anonfun$batchScriptedRunner$9(ScriptedTests.scala:148)
[error] 	at sbt.scriptedtest.ScriptedRunner.$anonfun$runAll$1(ScriptedTests.scala:467)
[error] 	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.flatmap2combiner(ParArray.scala:419)
[error] 	at scala.collection.parallel.ParIterableLike$FlatMap.leaf(ParIterableLike.scala:1074)
[error] 	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[error] 	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[error] 	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[error] 	at scala.collection.parallel.ParIterableLike$FlatMap.tryLeaf(ParIterableLike.scala:1070)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[error] 	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error] 	at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:389)
[error] 	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
[error] 	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync(Tasks.scala:379)
[error] 	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync$(Tasks.scala:379)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:440)
[error] 	at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult(Tasks.scala:423)
[error] 	at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult$(Tasks.scala:416)
[error] 	at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:60)
[error] 	at scala.collection.parallel.ParIterableLike$ResultMapping.leaf(ParIterableLike.scala:960)
[error] 	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[error] 	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[error] 	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[error] 	at scala.collection.parallel.ParIterableLike$ResultMapping.tryLeaf(ParIterableLike.scala:955)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[error] 	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error] 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[error] 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[error] 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[error] Caused by: java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src
[error] 	at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:83)
[error] 	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
[error] 	at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
[error] 	at sun.nio.fs.WindowsFileSystemProvider.createDirectory(WindowsFileSystemProvider.java:504)
[error] 	at java.nio.file.Files.createDirectory(Files.java:674)
[error] 	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
[error] 	at java.nio.file.Files.createDirectories(Files.java:767)
[error] 	at sbt.io.IO$.$anonfun$createDirectory$1(IO.scala:333)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:40)
[error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[error] 	at sbt.io.IO$.createDirectory(IO.scala:335)
[error] 	at sbt.io.IO$.copyImpl(IO.scala:835)
[error] 	at sbt.io.IO$.$anonfun$copy$1(IO.scala:826)
[error] 	at scala.Function$.$anonfun$tupled$1(Function.scala:79)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:273)
[error] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:273)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:266)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at sbt.io.IO$.copy(IO.scala:826)
[error] 	at sbt.io.IO$.copyDirectory(IO.scala:872)
[error] 	at sbt.io.IO$.copyDirectory(IO.scala:861)
[error] 	at sbt.io.IO$.copyDirectory(IO.scala:844)
[error] 	at sbt.scriptedtest.ScriptedTests.$anonfun$runBatchedTests$1(ScriptedTests.scala:245)
[error] 	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:273)
[error] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[error] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[error] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[error] 	at scala.collection.TraversableLike.map(TraversableLike.scala:273)
[error] 	at scala.collection.TraversableLike.map$(TraversableLike.scala:266)
[error] 	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
[error] 	at sbt.scriptedtest.ScriptedTests.runBatchTests$1(ScriptedTests.scala:240)
[error] 	at sbt.scriptedtest.ScriptedTests.runBatchedTests(ScriptedTests.scala:291)
[error] 	at sbt.scriptedtest.ScriptedTests.$anonfun$batchScriptedRunner$10(ScriptedTests.scala:148)
[error] 	at sbt.io.IO$.withTemporaryDirectory(IO.scala:490)
[error] 	at sbt.io.IO$.withTemporaryDirectory(IO.scala:500)
[error] 	at sbt.scriptedtest.ScriptedTests.$anonfun$batchScriptedRunner$9(ScriptedTests.scala:148)
[error] 	at sbt.scriptedtest.ScriptedRunner.$anonfun$runAll$1(ScriptedTests.scala:467)
[error] 	at scala.collection.parallel.mutable.ParArray$ParArrayIterator.flatmap2combiner(ParArray.scala:419)
[error] 	at scala.collection.parallel.ParIterableLike$FlatMap.leaf(ParIterableLike.scala:1074)
[error] 	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[error] 	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[error] 	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[error] 	at scala.collection.parallel.ParIterableLike$FlatMap.tryLeaf(ParIterableLike.scala:1070)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[error] 	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error] 	at java.util.concurrent.ForkJoinTask.doJoin(ForkJoinTask.java:389)
[error] 	at java.util.concurrent.ForkJoinTask.join(ForkJoinTask.java:719)
[error] 	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync(Tasks.scala:379)
[error] 	at scala.collection.parallel.ForkJoinTasks$WrappedTask.sync$(Tasks.scala:379)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.sync(Tasks.scala:440)
[error] 	at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult(Tasks.scala:423)
[error] 	at scala.collection.parallel.ForkJoinTasks.executeAndWaitResult$(Tasks.scala:416)
[error] 	at scala.collection.parallel.ForkJoinTaskSupport.executeAndWaitResult(TaskSupport.scala:60)
[error] 	at scala.collection.parallel.ParIterableLike$ResultMapping.leaf(ParIterableLike.scala:960)
[error] 	at scala.collection.parallel.Task.$anonfun$tryLeaf$1(Tasks.scala:53)
[error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] 	at scala.util.control.Breaks$$anon$1.catchBreak(Breaks.scala:67)
[error] 	at scala.collection.parallel.Task.tryLeaf(Tasks.scala:56)
[error] 	at scala.collection.parallel.Task.tryLeaf$(Tasks.scala:50)
[error] 	at scala.collection.parallel.ParIterableLike$ResultMapping.tryLeaf(ParIterableLike.scala:955)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute(Tasks.scala:153)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingTasks$WrappedTask.compute$(Tasks.scala:149)
[error] 	at scala.collection.parallel.AdaptiveWorkStealingForkJoinTasks$WrappedTask.compute(Tasks.scala:440)
[error] 	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189)
[error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[error] 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[error] 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[error] 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
[error] (scripted) java.io.IOException: Could not create directory C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src: java.nio.file.AccessDeniedException: C:\Users\appveyor\AppData\Local\Temp\1\sbt_5d0663fd\src
```

</details>

## expectations

Retry in case of AccessDeniedException?

## notes

https://github.com/sbt/io/commit/c2ff0018b370b080706ee9a8f162813be5f2b2ec

```diff
-        case _: IOException if count < 5 => create(count + 1)
+      catch { case _: IOException if Files.isDirectory(path) => },
```

This added `Files.isDirectory(path)` as condition to catching, so that's suspect.

--- a/io/src/main/scala/sbt/io/IO.scala
+++ b/io/src/main/scala/sbt/io/IO.scala
@@ -332,7 +332,7 @@ object IO {
     try Retry(
       try Files.createDirectories(path)
       catch { case _: IOException if Files.isDirectory(path) => },
-      excludedExceptions = classOf[FileAlreadyExistsException]
+      excludedExceptions = classOf[FileSystemException]
     )
     catch { case e: IOException => throw new IOException(failBase + ": " + e, e) }
     ()

